internal/handlers/billing/subscription.go

```diff
@@ -0,0 +1,393 @@
+	// Build limits for all subscription types
+	trialLimits := api.SubscriptionLimits{
+		MaxCustomerOrganizations: int64(
+			subscription.GetCustomersPerOrganizationLimit(types.SubscriptionTypeTrial),
+		),
+		MaxUsersPerCustomerOrganization: int64(
+			subscription.GetUsersPerCustomerOrganizationLimit(types.SubscriptionTypeTrial),
+		),
+		MaxDeploymentsPerCustomerOrg: int64(
+			subscription.GetDeploymentTargetsPerCustomerOrganizationLimit(types.SubscriptionTypeTrial),
+		),
+	}
+
+	starterLimits := api.SubscriptionLimits{
+		MaxCustomerOrganizations: int64(
+			subscription.GetCustomersPerOrganizationLimit(types.SubscriptionTypeStarter),
+		),
+		MaxUsersPerCustomerOrganization: int64(
+			subscription.GetUsersPerCustomerOrganizationLimit(types.SubscriptionTypeStarter),
+		),
+		MaxDeploymentsPerCustomerOrg: int64(
+			subscription.GetDeploymentTargetsPerCustomerOrganizationLimit(types.SubscriptionTypeStarter),
+		),
+	}
+
+	proLimits := api.SubscriptionLimits{
+		MaxCustomerOrganizations: int64(
+			subscription.GetCustomersPerOrganizationLimit(types.SubscriptionTypePro),
+		),
+		MaxUsersPerCustomerOrganization: int64(
+			subscription.GetUsersPerCustomerOrganizationLimit(types.SubscriptionTypePro),
+		),
+		MaxDeploymentsPerCustomerOrg: int64(
+			subscription.GetDeploymentTargetsPerCustomerOrganizationLimit(types.SubscriptionTypePro),
+		),
+	}
+
+	enterpriseLimits := api.SubscriptionLimits{
+		MaxCustomerOrganizations: int64(
+			subscription.GetCustomersPerOrganizationLimit(types.SubscriptionTypeEnterprise),
+		),
+		MaxUsersPerCustomerOrganization: int64(
+			subscription.GetUsersPerCustomerOrganizationLimit(types.SubscriptionTypeEnterprise),
+		),
+		MaxDeploymentsPerCustomerOrg: int64(
+			subscription.GetDeploymentTargetsPerCustomerOrganizationLimit(types.SubscriptionTypeEnterprise),
+		),
+	}
+
+	info := &api.SubscriptionInfo{
+		SubscriptionType:                       org.SubscriptionType,
+		SubscriptionEndsAt:                     org.SubscriptionEndsAt.Format("2006-01-02"),
+		SubscriptionExternalID:                 org.StripeSubscriptionId,
+		SubscriptionCustomerOrganizationQty:    org.SubscriptionCustomerOrganizationQty,
+		SubscriptionUserAccountQty:             org.SubscriptionUserAccountQty,
+		CurrentUserAccountCount:                usage.userAccountCount,
+		CurrentCustomerOrganizationCount:       usage.customerOrganizationCount,
+		CurrentMaxUsersPerCustomer:             usage.maxUsersPerCustomer,
+		CurrentMaxDeploymentTargetsPerCustomer: usage.maxDeploymentTargetsPerCustomer,
+		TrialLimits:                            trialLimits,
+		StarterLimits:                          starterLimits,
+		ProLimits:                              proLimits,
+		EnterpriseLimits:                       enterpriseLimits,
+	}
+
+	return info, nil
+}
+
+// currentUsageCounts represents the current usage counts for an organization
+type currentUsageCounts struct {
+	userAccountCount                int64
+	customerOrganizationCount       int
```

Comment:
is there a reason to use a different type for this one property? I think they should be all the same